executor/write.go

```diff
@@ -156,7 +157,17 @@ func updateRecord(ctx context.Context, sctx sessionctx.Context, h int64, oldData
 		if sc.DupKeyAsWarning {
 			newHandle, err = t.AddRecord(sctx, newData, table.IsUpdate, table.SkipHandleCheck, table.WithCtx(ctx))
 		} else {
+			txn, err1 := sctx.Txn(true)
+			if err1 != nil {
+				return false, false, 0, err1
+			}
+			// If there are primary keys or unique indices, we have to check TiKV to ensure their uniqueness.
+			// The PresumeKeyNotExists option could delay the check to improve performance.
+			if !sctx.GetSessionVars().ConstraintCheckInPlace {
+				txn.SetOption(kv.PresumeKeyNotExists, nil)
```

Comment:
It should only work for auto committed SQLs. Otherwise, it will cause compatibility issues.

Reply:
Fixed